frontend/js/src/mbid-mapping/MbidMappingModal.tsx

```diff
@@ -0,0 +1,151 @@
+
+  constructor(props: MbidMappingModalProps) {
+    super(props);
+    this.state = { recordingMBID: "" };
+  }
+
+  submitMbidMapping = async () => {
+    const { recordingToMap, newAlert, onSuccessfulPin } = this.props;
+    const { recordingMBID } = this.state;
+    const { APIService, currentUser } = this.context;
+
+    if (recordingToMap && currentUser?.auth_token) {
+      const recordingMSID = _get(
+        recordingToMap,
+        "track_metadata.additional_info.recording_msid"
+      );
+
+      try {
+        const response = await APIService.submitMbidMapping(
+          currentUser.auth_token,
+          recordingMSID,
+          recordingMBID
+        );
+        const { data } = response;
+      } catch (error) {
+        this.handleError(error, "Error while pinning track");
+        return;
+      }
+
+      newAlert(
+        "success",
+        `You mapped a track!`,
+        `${getArtistName(recordingToMap)} - ${getTrackName(recordingToMap)}`
+      );
+      this.setState({ recordingMBID: "" });
+
+      // if (onSuccessfulPin) {
+      //   onSuccessfulPin(newPin);
+      // }
+    }
+  };
+
+  handleMbidInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    const input = event.target.value;
+    this.setState({ recordingMBID: input });
+  };
+
+  handleError = (error: string | Error, title?: string): void => {
+    const { newAlert } = this.props;
+    if (!error) {
+      return;
+    }
+    newAlert(
+      "danger",
+      title || "Error",
+      typeof error === "object" ? error.message : error
+    );
+  };
+
+  render() {
+    const { recordingToMap } = this.props;
+    if (!recordingToMap) {
+      return null;
+    }
+    const { recordingMBID } = this.state;
+
+    return (
+      <div
+        className="modal fade"
+        id="MapToMusicBrainzRecordingModal"
+        tabIndex={-1}
+        role="dialog"
+        aria-labelledby="MbidMappingModalLabel"
+        data-backdrop="static"
+      >
+        <div className="modal-dialog" role="document">
+          <form className="modal-content">
+            <div className="modal-header">
+              <button
+                type="button"
+                className="close"
+                data-dismiss="modal"
+                aria-label="Close"
+              >
+                <span aria-hidden="true">&times;</span>
+              </button>
+              <h4 className="modal-title" id="MbidMappingModalLabel">
+                Set a Recording MBID for this Listen
+              </h4>
+            </div>
+            <div className="modal-body">
+              <div className="form-group">
+                <input
+                  type="text"
+                  value={recordingMBID}
+                  className="form-control"
+                  id="recording-mbid"
+                  name="recording-mbid"
+                  onChange={this.handleMbidInputChange}
+                />
```

Comment:
Future addition to this could be to integrate the artist/track search that we have on the playlist create page so that users can search directly from here

Reply:
added to LB-1179